Makefile

```diff
@@ -0,0 +1,40 @@
+CXX          ?= g++
+CXXOPTS      ?= -g -Wall -Werror -std=c++11 -Iinclude/
+DEBUG_OPTS   ?= -fprofile-arcs -ftest-coverage -O0 -g -fPIC
+RELEASE_OPTS ?= -O3
+
+all: test
+
+# Release libraries
+release/libsimhash.o: release/simhash.o release/permutation.o
+	mkdir -p release
```

Comment:
You could just make `release` a target, which does this, and then have a dependency. Same thing for your `debug` directory.